src/main/java/de/turing85/quarkus/vertx/filter/etag/proxy/ETagInterceptor.java

```diff
@@ -47,34 +49,29 @@ private static Future<Void> handleETag(ProxyContext context, String eTag) {
   }
 
   private Future<Void> generateAndHandleETag(ProxyContext context) {
-    try {
-      final HashingWriteStream write = new HashingWriteStream("MD5", vertx);
-      // @formatter:off
-      return context.response().getBody().stream()
-          .pipeTo(write)
-          .compose(unused -> handleETagFromWriter(context, write))
-          .onComplete(unused -> write.close());
-      // @formatter:on
-    } catch (Exception e) {
-      return Future.failedFuture(e);
-    }
+    // @formatter:off
+    ReadStream<Buffer> bodyStream = context.response().getBody().stream().pause();
```

Comment:
Can the call to `pause()` be moved into the `process`-method of the helper?

Reply:
You can't move the .pause() call inside process. Otherwise, the backend response might be consumed completely before the proxy sets its handler (and throws the infamous Request has already been read)